processor/src/main/java/org/mapstruct/ap/internal/model/source/ForgedMethod.java

```diff
@@ -303,4 +306,14 @@ public int hashCode() {
         result = 31 * result + ( name != null ? name.hashCode() : 0 );
         return result;
     }
+
+    private Parameter determineMappingTargetParameter(Iterable<Parameter> parameters) {
```

Comment:
Just a not for the future, we can extract this construct to some util. We use it in multiple places 😄